vmware_exporter/vmware_exporter.py

```diff
@@ -32,172 +44,223 @@
 from prometheus_client.core import GaugeMetricFamily
 from prometheus_client import CollectorRegistry, generate_latest
 
-from .helpers import batch_fetch_properties, get_bool_env
+from .helpers import batch_fetch_properties, get_bool_env, get_unverified_session
 from .defer import parallelize, run_once_property
 
 
 class VmwareCollector():
 
-    def __init__(self, host, username, password, collect_only, specs_size, ignore_ssl=False):
+    def __init__(
+        self,
+        host,
+        username,
+        password,
+        collect_only,
+        specs_size,
+        fetch_custom_attributes=True,
+        ignore_ssl=False,
+        fetch_tags=True
+    ):
+
         self.host = host
         self.username = username
         self.password = password
         self.ignore_ssl = ignore_ssl
         self.collect_only = collect_only
         self.specs_size = int(specs_size)
 
+        # Custom Attributes
+        # flag to wheter fetch custom attributes or not
+        self.fetch_custom_attributes = fetch_custom_attributes
+        # vms, hosts and datastores custom attributes must be stored by their moid
+        self._vmsCustomAttributes = {}
+        self._hostsCustomAttributes = {}
+        self._datastoresCustomAttributes = {}
+
+        # Tags
+        # flag to wheter fetch tags or not
+        self.fetch_tags = fetch_tags
+
+        # label names and ammount will be needed later to insert labels from custom attributes
+        self._labelNames = {
```

Comment:
I like what you did here.

Reply:
Oh! Thanks! I thought it would like more synthetic and less places to change if needed.